pkg/formatted/color.go

```diff
@@ -16,19 +16,64 @@ package formatted
 
 import (
 	"io"
+	"sync"
 
 	"github.com/fatih/color"
 )
 
+var (
+	mux       sync.Mutex
+	allColors = []color.Attribute{
+		color.FgHiGreen,
+		color.FgHiYellow,
+		color.FgHiBlue,
+		color.FgHiMagenta,
+		color.FgHiCyan,
+		color.FgHiWhite,
+	}
+)
+
+type RainbowColors struct {
+	current int
+	cached  map[string]color.Attribute
+	all     []color.Attribute
+}
+
+func NewRainbowColors() *RainbowColors {
+	return &RainbowColors{
+		current: 0,
+		cached:  make(map[string]color.Attribute),
+		all:     allColors,
+	}
+}
+
+func (rc *RainbowColors) Get(label string) *color.Color {
+	mux.Lock()
+	defer mux.Unlock()
+
+	if _, ok := rc.cached[label]; !ok {
+		rc.cached[label] = rc.all[rc.current]
+		rc.current++
+		if rc.current >= len(rc.all) {
```

Comment:
nit: usually this is done using `rc.current = (rc.current +1) % len(rc.all)` without having the need for an `if`